gst/nnstreamer/tensor_filter/tensor_filter.c

```diff
@@ -909,47 +984,121 @@ gst_tensor_filter_transform (GstBaseTransform * trans,
       meta.format = _NNS_TENSOR_FORMAT_FLEXIBLE;
 
       flex_mem = gst_memory_new_wrapped (0,
-          out_tensors[i].data, out_tensors[i].size, 0,
-          out_tensors[i].size, out_tensors[i].data, g_free);
+          trans_data->out_tensors[i].data, trans_data->out_tensors[i].size, 0,
+          trans_data->out_tensors[i].size, trans_data->out_tensors[i].data,
+          g_free);
 
-      out_mem[i] = gst_tensor_meta_info_append_header (&meta, flex_mem);
+      trans_data->out_mem[i] =
+          gst_tensor_meta_info_append_header (&meta, flex_mem);
       gst_memory_unref (flex_mem);
-    } else if (allocate_in_invoke) {
+    } else if (trans_data->allocate_in_invoke) {
       /* prepare memory block if successfully done */
-      out_mem[i] = mem = gst_tensor_filter_get_wrapped_mem (self,
-          out_tensors[i].data, out_tensors[i].size);
+      trans_data->out_mem[i] = mem = gst_tensor_filter_get_wrapped_mem (self,
+          trans_data->out_tensors[i].data, trans_data->out_tensors[i].size);
 
-      if (out_flexible) {
+      if (trans_data->out_flexible) {
         /* prepare new memory block with meta */
-        out_mem[i] = gst_tensor_meta_info_append_header (&out_meta[i], mem);
+        trans_data->out_mem[i] =
+            gst_tensor_meta_info_append_header (&trans_data->out_meta[i], mem);
         gst_memory_unref (mem);
       }
     }
 
     /* append the memory block to outbuf */
-    gst_tensor_buffer_append_memory (outbuf, out_mem[i],
+    gst_tensor_buffer_append_memory (trans_data->outbuf, trans_data->out_mem[i],
         gst_tensors_info_get_nth_info (&prop->output_meta, i));
   }
 
   return GST_FLOW_OK;
+}
+
+/**
+ * @brief non-ip transform. required vmethod of GstBaseTransform.
+ */
+static GstFlowReturn
+gst_tensor_filter_transform (GstBaseTransform * trans,
+    GstBuffer * inbuf, GstBuffer * outbuf)
+{
+  GstTensorFilter *self = GST_TENSOR_FILTER_CAST (trans);
+  GstTensorFilterPrivate *priv = &self->priv;
+  GstTensorFilterProperties *prop = &priv->prop;
+  guint i;
+  gint invoke_res = -1;
+  gboolean need_profiling;
+  GstFlowReturn retval = GST_FLOW_OK;
+
+  FilterTransformData trans_data = { 0, };
```

Comment:
need allocating memory for trans_data? not stack.

Reply:
Changed to allocate mem. Thanks.